Library/Homebrew/api/cask.rb

```diff
@@ -20,16 +20,34 @@ def fetch_source(token, git_head: nil)
           Homebrew::API.fetch_homebrew_cask_source token, git_head: git_head
         end
 
+        sig { returns(T::Boolean) }
+        def prepare_json_data
+          json_casks, updated = Homebrew::API.fetch_json_api_file "cask.json",
+                                                                  target: HOMEBREW_CACHE_API/"cask.json"
+
+          @all_casks = json_casks.to_h do |json_cask|
+            [json_cask["token"], json_cask.except("token")]
+          end
+
+          updated
+        end
+        private :prepare_json_data
+
         sig { returns(Hash) }
         def all_casks
-          @all_casks ||= begin
-            json_casks = Homebrew::API.fetch_json_api_file "cask.json",
-                                                           target: HOMEBREW_CACHE_API/"cask.json"
-
-            json_casks.to_h do |json_cask|
-              [json_cask["token"], json_cask.except("token")]
-            end
+          unless defined?(@all_casks)
+            json_updated = prepare_json_data
+            write_names(regenerate: json_updated)
           end
+
+          @all_casks
+        end
+
+        sig { params(regenerate: T::Boolean).void }
+        def write_names(regenerate: false)
+          prepare_json_data unless defined?(@all_casks)
```

Comment:
I think it'd be reasonable to just call `all_casks` here and merge `prepare_json_data` and `all_casks` into a single method.